include/hip/hcc_detail/functional_grid_launch.hpp

```diff
@@ -202,22 +203,24 @@ hipError_t hipExtLaunchMultiKernelMultiDevice(hipLaunchParams* launchParamsList,
 template <typename F>
 inline
 __attribute__((visibility("hidden")))
-hipError_t hipLaunchCooperativeKernel(F f, dim3 gridDim, dim3 blockDimX, void** kernelParams,
-                unsigned int sharedMemBytes, hipStream_t stream) {
-
+hipError_t hipLaunchCooperativeKernel(F f, dim3 gridDim, dim3 blockDim,
+                                      void** args, size_t sharedMem,
+                                      hipStream_t stream) {
     hip_impl::hip_init();
     auto& ps = hip_impl::get_program_state();
```

Comment:
we can always get program_state in internal functions do we really need to get it in here?

Reply:
@SarbojitAMD Yes, the hip_impl::get_program_state() is intentionally used here to get the correct program state during lazy binding. Please see [PR#1698](https://github.com/ROCm-Developer-Tools/HIP/pull/1698) for more details.